master/reference/typha/configuration.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Configuring Typha
+canonical_url: 'https://docs.projectcalico.org/v3.1/reference/typha/configuration'
+---
+
+Configuration for Typha is read from one of two possible locations, in
+order, as follows.
+
+1.  Environment variables, prefixed with `TYPHA_`.
+
+2.  The Typha configuration file.  The path to this file defaults to
+    `/etc/calico/typha.cfg` but can be overridden using the `-c` or
+    `--config-file` options on the command line.
+
+The value of any configuration parameter is the value read from the
+*first* location containing a value. For example, if an environment variable
+contains a value, it takes precedence.
+
+If not set in any of these locations, most configuration parameters have
+defaults, and it should be rare to have to explicitly set them.
+
+The full list of parameters which can be set is as follows.
+
+#### General configuration
+
+| Configuration parameter           | Environment variable                    | Description  | Schema |
+| --------------------------------- | --------------------------------------- | -------------| ------ |
+| `DatastoreType`                   | `TYPHA_DATASTORETYPE`                   | The datastore that Typha should read endpoints and policy information from. [Default: `etcdv3`] | `etcdv3`, `kubernetes`|
+| `HealthEnabled`                   | `TYPHA_HEALTHENABLED`                   | When enabled, exposes Typha health information via an http endpoint. | boolean |
+| `LogFilePath`                     | `TYPHA_LOGFILEPATH`                     | The full path to the Typha log. Set to `none` to disable file logging. [Default: `/var/log/calico/typha.log`] | string |
+| `LogSeverityFile`                 | `TYPHA_LOGSEVERITYFILE`                 | The log severity above which logs are sent to the log file. [Default: `Info`] | `Debug`, `Info`, `Warning`, `Error`, `Fatal` |
+| `LogSeverityScreen`               | `TYPHA_LOGSEVERITYSCREEN`               | The log severity above which logs are sent to the stdout. [Default: `Info`] | `Debug`, `Info`, `Warning`, `Error`, `Fatal` |
+| `LogSeveritySys`                  | `TYPHA_LOGSEVERITYSYS`                  | The log severity above which logs are sent to the syslog. Set to `""` for no logging to syslog. [Default: `Info`] | `Debug`, `Info`, `Warning`, `Error`, `Fatal` |
+| `PrometheusGoMetricsEnabled`      | `TYPHA_PROMETHEUSGOMETRICSENABLED`      | Set to `false` to disable Go runtime metrics collection, which the Prometheus client does by default. This reduces the number of metrics reported, reducing Prometheus load. [Default: `true`]  | boolean |
+| `PrometheusMetricsEnabled`        | `TYPHA_PROMETHEUSMETRICSENABLED`        | Set to `true` to enable the experimental Prometheus metrics server in Typha. [Default: `false`] | boolean |
```

Comment:
I don't think we need to call prometheus metrics experimental?

Reply:
I've pushed a markup to delete that 'experimental' for both Felix and Typha.